executor/runtime/docker/docker_linux.go

```diff
@@ -84,6 +84,47 @@ func setupScheduler(cred ucred) error {
 	return nil
 }
 
+func isCgroup2HybridMode() bool {
+	var st unix.Statfs_t
+	err := unix.Statfs(hybridMountpoint, &st)
+	if err != nil {
+		if os.IsNotExist(err) {
+			// ignore the "not found" error
+			return false
+		}
+		panic(fmt.Sprintf("cannot statfs cgroup root: %s", err))
+	}
+	return st.Type == unix.CGROUP2_SUPER_MAGIC
+}
+
+// move container's pid 1 to unified cgroup hierarchy to enable eBPF hooks
+func movePid1ToUnifiedController(cred ucred) error {
+	if !isCgroup2HybridMode() {
+		return nil
+	}
+	cgroupPath := filepath.Join("/proc/", strconv.Itoa(int(cred.pid)), "cgroup")
+	cgroupsInfo, err := cgroups.ParseCgroupFile(cgroupPath)
+	if err != nil {
+		logrus.WithError(err).Error("Could not read container cgroups")
+		return err
+	}
+	// Find path to the systemd owned cgroup
+	if controllerPath, ok := cgroupsInfo["name=systemd"]; ok {
+
+		fsPath := filepath.Join(sysFsCgroup, "unified", controllerPath)
```

Comment:
Didn't we talk about having an assertion to ensure that this is under the titus-executor's cgroup2?

Reply:
added a check